lib/fe.ex

```diff
@@ -15,4 +15,23 @@ defmodule FE do
   """
   @spec const(a) :: (any -> a) when a: var
   def const(a), do: fn _ -> a end
+  
+  @doc """
+  Transforms a n-ary function in an unary one that accepts its first
+  argument and returns a function expecting the next one, successively
+  """
+  @spec curry((... -> any())) :: (any() -> any())
+  def curry(fun) do
+    case :erlang.fun_info(fun, :arity) do
+      {_, 0} -> raise ArgumentError, "0-ary functions cannot be curried"
```

Comment:
I'm wondering if it wouldn't be cleaner if we always returned a zero argument function from the last step always.

I can see the following pros:
- no special case and raising for zero argument functions in `curry`;
- more in line with languages that have currying built-in (laziness);
- we should be able to improve the vague spec of this function.